src/index.ts

```diff
@@ -1,179 +1,221 @@
+import BN = require('bn.js')
+import rlp = require('rlp')
 const createKeccakHash = require('keccak')
 const secp256k1 = require('secp256k1')
 const assert = require('assert')
-const rlp = require('rlp')
-const BN = require('bn.js')
 const createHash = require('create-hash')
 const Buffer = require('safe-buffer').Buffer
-Object.assign(exports, require('ethjs-util'))
+const ethjsUtil = require('ethjs-util')
+Object.assign(exports, ethjsUtil)
+
+// Types
+export interface ECDSASignature {
+  v: number
+  r: Buffer
+  s: Buffer
+}
+
+export type BufferableArray = number[] | ArrayBuffer | SharedArrayBuffer
+
+export interface Arrayable {
+  toArray(): BufferableArray
+}
+
+export type Bufferable =
+  | Buffer
+  | string
+  | number
+  | null
+  | undefined
+  | BN
+  | BufferableArray
+  | Arrayable
```

Comment:
Types and interfaces in the `RLP` implementation have gotten their own [types.ts](https://github.com/ethereumjs/rlp/blob/master/src/types.ts) file. I'm not deep enough into `TypeScript` yet to judge what is better / best practice. So just asking: what is your reasoning here to do it this way?

Reply:
Have no strong feeling about it, if you want just leave as is. Some reorg likely also makes sense on this library - see also e.g. https://github.com/ethereumjs/ethereumjs-util/pull/38. We should eventually do this in a second round though, think there is enough to concentrate on here.